### PR TITLE
MCR-2397 remove old hibernate code

### PR DESCRIPTION
**Pull request by yagee-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2397).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on c929061578fca81fd3d83b3fe490fde43ac68aa5! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6992-pull_request-1229

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 9f453c2b9b6b5402125bca4d92b7379e459463d8! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6994-pull_request-1229

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on b77215c725547e874b3b06bfdec13a215e03b2b4! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6996-pull_request-1229

**Comment by kkrebs** [comment]
please rebase and solve conflicts

File: mycore-base/src/main/java/org/mycore/common/MCRSession.java [hunk 1]
```diff
@@ -160,7 +159,8 @@ public void close() {
     MCRSession() {
         userInformation = guestUserInformation;
         setCurrentLanguage(MCRConfiguration2.getString("MCR.Metadata.DefaultLang").orElse(MCRConstants.DEFAULT_LANG));
-        dataBaseAccess = MCRHIBConnection.isEnabled();
+        dataBaseAccess = MCRConfiguration2.getBoolean("MCR.Persistence.Database.Enable").orElse(true)
```
Comment (sebhofmann) [inline mycore-base/src/main/java/org/mycore/common/MCRSession.java]: I think this maybe needs to be 
```
        dataBaseAccess = MCRConfiguration2.getBoolean("MCR.Persistence.Database.Enable").orElse(true)
            && TRANSACTION_SERVICE_LOADER.stream().map(ServiceLoader.Provider::get)
                .anyMatch(MCRPersistenceTransaction::isReady);
```

**Review by sebhofmann — APPROVED** [review]

**Review by rsteph-de — APPROVED** [review]
looks good for me, ready for merge

**Pull request merged**